GetStream/stream-chat-android
# Fix URL encoding on various android/java versions

Closes #3003 

### 🎯 Goal

After researching into a user-reported issue, we've found that our encoding logic wasn't working as the property used for the encoding was faulty.

### 🛠 Implementation details

Instead of using `encoding.toString()` we used `encoding.name()` as in different Java/JVM environments (e.g. older devices) `toString()` will not return the proper value.

### 🧪 Testing

1. Run the UI comp sample app on an API 21-25 device _without_ this fix and attempt to open a PDF file. The app will crash or it will throw an error saying it cannot preview a PDF.
2. Run the sample app _with_ this fix and it should work fine.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![giphy (1)](https://user-images.githubusercontent.com/17215808/152301809-686fb68e-7653-445c-bc36-abf7ecf6816b.gif)
